### PR TITLE
feat: Implement Create Todo with form and submition

### PR DESCRIPTION
**Pull request by RezaBG**
- install react-hook form
- create a component that displays a form for creating a todo
- display the create form on the home page
- add a button at the top of the list of todos that is labeled 'Add todo'
- Hide the create form from the homepage
- When the create button is clicked, display the create form
- Implement the todo creation: when the submit button in the form is clicked, a todo should be created

Jira ticket:
https://boir.atlassian.net/browse/KAN-46?atlOrigin=eyJpIjoiNGY4OTFkYTVjNTBiNDI3MThhOGY3ZjM5N2E2NGMxOWMiLCJwIjoiaiJ9

### CODE REVIEW

**Review by gnerkus — CHANGES_REQUESTED** [review]
Great addition and use of `react-hook-form`.

I've left a few required comments:
- ⚠️ -> required
- ℹ️ -> recommended

File: src/App.tsx [hunk 1]
```diff
@@ -20,22 +24,25 @@ function App() {
...
   const [showForm, setShowForm] = useState(false);
 
-  const handleSubmitTodo = async (newTodo: TodoData) => {
-    try {
-      const response = await axiosInstance.post<TodoResponse>("todos", newTodo);
-      console.log("Todo created:", response.data);
+  const mutation = useMutation({
+    mutationFn: async (newTodo: TodoData) => {
+      await axiosInstance.post<TodoResponse>("todos", newTodo);
       await queryClient.invalidateQueries({ queryKey: ["todos"] });
```
Comment (gnerkus) [inline src/App.tsx:30]: > Mutations automatically invalidate the related queries
This was wrong on my part so you're correct to leave in the `invalidateQueries` call.

⚠️  This should move into an `onSuccess` callback (https://tkdodo.eu/blog/mastering-mutations-in-react-query#invalidation) :

```
useMutation({
  mutationFn: async () => ...,
  onSuccess: () => {
    queryClient.invalidateQueries()
  }
})
```

**Review by gnerkus — COMMENTED** [review]
Left a comment

File: src/App.tsx [hunk 2]
```diff
@@ -20,22 +24,25 @@ function App() {
   const queryClient = useMemo(() => new QueryClient(), []);
   const [showForm, setShowForm] = useState(false);
 
-  const handleSubmitTodo = async (newTodo: TodoData) => {
-    try {
-      const response = await axiosInstance.post<TodoResponse>("todos", newTodo);
-      console.log("Todo created:", response.data);
+  const mutation = useMutation({
```
Comment (gnerkus) [inline src/App.tsx:27]: ℹ️ A little re-organization might help here.

TL;DR:
1. Name the `mutation` e.g `createTodoMutation`
2. move `createTodoMutation` into a separate file e.g _mutations.ts_
3. move the `handleSubmitTodo` out of `<App />`
4. remove the `onSubmit` prop of `<CreateTodo />`
5. import the `createTodoMutation` mutation and use in `<CreateTodo />`

## Explanation
### Name the `mutation` e.g `createTodoMutation`
This will help identify the mutation in the future as there will be more mutations as the app grows.

### move `createTodoMutation` into a separate file e.g _mutations.ts_
This will allow components to not have to depend on `<App />`; it will allow any part of the application access to the mutations. 

### move the `handleSubmitTodo` out of `<App />`
The also removes the dependency of components on `<App />`. The `<App />`'s job is just to render; only the components should know how they work internally. If the dependency is not removed then we'll end up defining all the mutations in `<App />`

### remove the `onSubmit` prop of `<CreateTodo />`
If you move the mutation into its own file, you will no longer need the prop.

### import the `createTodoMutation` mutation and use in `<CreateTodo />`
`<CreateTodo />` should not have direct access to `createTodoMutation` without relying on `<App />` to pass it down via a prop.

Comment (gnerkus) [reply src/App.tsx:27]: ⚠️ Please, move the `fetchTodos` function into the _api.ts_ file

https://github.com/RezaBG/TUDU_FE/pull/3/files#diff-faf663d4dd497fd71dff9adbed49bf1f75c297ed67517dbd6a049f90b345b52eR11-R14

**Review by gnerkus — APPROVED** [review]
LGTM 👍 

Thanks a lot for implementing the recommendation!
